pkg/controller/k8/cluster.go

```diff
@@ -199,9 +201,30 @@ func (k *Cluster) UpdateK8Object(ctx context.Context, object runtime.Object) err
 
 func (k *Cluster) UpdateStatus(ctx context.Context, object runtime.Object) error {
 	objectCopy := object.DeepCopyObject()
-
 	err := k.client.Status().Update(ctx, objectCopy)
 	if err != nil {
+		if errors.IsInvalid(err) {
+			// This is a Kubernetes bug that has been fixed in k8s 1.15
+			// https://github.com/kubernetes/kubernetes/pull/78713
+			// The bug prevents status sub-resources from being updated when
+			// the stored version of the CRD changes
+			// Example of error:
+			// K8s object update failed FlinkApplication.flink.k8s.io "operator-test-app" is invalid:
+			// apiVersion: Invalid value: "flink.k8s.io/v1beta1": must be flink.k8s.io/v1beta1
+			// app_name=operator-test-app ns=default phase=Running src="cluster.go:209"
+			// This should only ever be encountered once (per application)
+			// when a new CRD version is deployed and an older version of the application exists
+			// As a workaround, we try to update the entire resource instead of only the status
+			// TODO Remove this block when we upgrade to k8s 1.15
```

Comment:
I kept this around to help future CRD upgrades (though no longer required for this PR because there's no CRD upgrade anymore). Let me know if you don't want this here.